data/lib/core/player.lua

```diff
@@ -737,3 +727,46 @@ function Player.disableLoginMusic(self)
 	msg:delete()
 	return true
 end
+
+local blessFlags = {
+	[1] = BLESS_TYPE_WISDOM_OF_SOLITUDE,
+	[2] = BLESS_TYPE_SPARK_OF_PHOENIX,
+	[3] = BLESS_TYPE_FIRE_OF_THE_SUNS,
+	[4] = BLESS_TYPE_SPIRITUAL_SHIELD,
+	[5] = BLESS_TYPE_THE_EMBRACE_OF_THE_WORLD
+}
+
+function Player.getBlessings(self)
+	local n = 0
+	local flags = 0
+	for i = 1, SERVER_BLESSINGS_COUNT do
+		if self:hasBlessing(i) then
+			n = n + 1
+			flags = flags + blessFlags[i]
+		end
+	end
+	return n, flags
+end
+
+local function getBlessingStatus(n)
+	if n >= SERVER_BLESSINGS_COUNT then
+		return 3
+	elseif n > 0 then
+		return 2
+	else
+		return 1
+	end
+end
```

Comment:
Create var for ​​1, 2 and 3 values like

```suggestion
local function getBlessingStatus(n)
	if n >= SERVER_BLESSINGS_COUNT then
		return BLESSINGS_STATUS_FULL
	elseif n > 0 then
		return BLESSINGS_STATUS_PARTIAL
	end
	return BLESSINGS_STATUS_NONE
end
```